Samples/Mutation_Files/crossovermoff.py

```diff
@@ -0,0 +1,45 @@
+import random
+
+def single_point_crossover(timetable1, timetable2, num_offspring=4):
+    # Ensure both parents have the same length
+    assert len(timetable1) == len(timetable2), "Parents must have the same length"
+    
+    offspring_list = []
+    
+    for _ in range(num_offspring // 2):
+        # Choose a random crossover point
+        crossover_point = random.randint(1 , len(timetable1) - 1)
+        
+        # Generate two offspring by swapping genes at the crossover point
+        offspring1 = timetable1[:crossover_point] + timetable2[crossover_point:]
+        offspring2 = timetable2[:crossover_point] + timetable1[crossover_point:]
+        
+        offspring_list.append(offspring1)
+        offspring_list.append(offspring2)
```

Comment:
There is no verification for validity of the time table  it may lead to invalid  time table